pkg/minishift/openshift/service_test.go

```diff
@@ -0,0 +1,133 @@
+	expectString := "NAME\nmyproject\ntest"
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	got, err := getProjects()
+
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	expected := strings.Split(expectString, "\n")[1:]
+	if !reflect.DeepEqual(got, expected) {
+		t.Fatalf("Expected: %+v, Got: %+v", expected, got)
+	}
+}
+
+func Test_getServiceSpec(t *testing.T) {
+	setup()
+	defer teardown()
+
+	service := "foo"
+	namespace := "bar"
+	https := false
+
+	fakeRunner := test.NewFakeRunner(t)
+	runner = fakeRunner.Runner
+
+	args := fmt.Sprintf("get projects --config=%s %s", systemKubeConfigPath, ProjectsCustomCol)
+	expectString := fmt.Sprintf("NAME\n%s", namespace)
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	args = fmt.Sprintf("get route/%s -n %s --config=%s %s", service, namespace, systemKubeConfigPath, URLCustomCol)
+	expectString = "URL\nfoo.bar.xip.io"
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	got, err := GetServiceSpec(service, namespace, https)
+
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	expected := "http://" + strings.Split(expectString, "\n")[1]
+
+	if https {
+		expected = "https://" + strings.Split(expectString, "\n")[1]
+	}
+
+	if got != expected {
+		t.Fatalf("Expected: %+v, Got: %+v", expected, got)
+	}
+
+}
+
+func Test_getServiceSpecs(t *testing.T) {
+	setup()
+	defer teardown()
+
+	namespace := "bar"
+
+	fakeRunner := test.NewFakeRunner(t)
+	runner = fakeRunner.Runner
+
+	args := fmt.Sprintf("get projects --config=%s %s", systemKubeConfigPath, ProjectsCustomCol)
+	expectString := fmt.Sprintf("NAME\n%s", namespace)
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	args = fmt.Sprintf("get svc -o json --config=%s -n %s", systemKubeConfigPath, namespace)
+	expectString = testdata.ServiceSpec
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	args = fmt.Sprintf("get route -o json --config=%s -n %s", systemKubeConfigPath, namespace)
+	expectString = testdata.RouteSpec
+	fakeRunner.ExpectAndReturn(args, expectString)
+
+	got, err := GetServiceSpecs(namespace)
+
+	expected := []ServiceSpec{{Namespace: namespace, Name: "guestbook-v1-np", URL: []string(nil), NodePort: "32740", Weight: []string(nil)},
+		{Namespace: namespace, Name: "guestbook-v2-np", URL: []string(nil), NodePort: "30485", Weight: []string(nil)},
+		{Namespace: namespace, Name: "guestbook-v1", URL: []string{"http://guestbook-v1-3002-myproject.192.168.64.2.nip.io", "http://guestbook-myproject.192.168.64.2.nip.io", "http://guestbook-v1-myproject.192.168.64.2.nip.io"}, NodePort: "", Weight: []string{"", "50%", ""}},
+		{Namespace: namespace, Name: "guestbook-v2", URL: []string{"http://guestbook-v2-myproject.192.168.64.2.nip.io", "http://guestbook-v2-3002-myproject.192.168.64.2.nip.io", "http://guestbook-myproject.192.168.64.2.nip.io"}, NodePort: "", Weight: []string{"", "", "50%"}}}
+
+	if err != nil {
+		t.Fatalf("%+v", err)
+	}
+
+	if !reflect.DeepEqual(got, expected) {
+		t.Fatalf("\nExpected: %+v\nGot: %+v", expected, got)
```

Comment:
Here is why test case failing because `string{"http://guestbook-v2-myproject.192.168.64.2.nip.io", "http://guestbook-v2-3002-myproject.192.168.64.2.nip.io", "http://guestbook-myproject.192.168.64.2.nip.io"}` this is something part of map value when iterate so I need to find a way either compare bits manually or any smart way.